Linker fails when calling `hb_face_create` with Windows GNU Toolchain

When trying to build an executable that uses `harfbuzz-sys` on Windows with the GNU toolchain, calling `hb_face_create` results in the linker failing with the following error:

```
error: failed to run custom build command for `harfbuzz-sys v0.1.15`
process didn't exit successfully: `D:\programming\git\derin\target\debug\build\harfbuzz-sys-d6fc439b7abe6289\build-script-build` (exit code: 101)
--- stdout
running: "cmake" "C:\\Users\\ossi\\.cargo\\registry\\src\\github.com-1ecc6299db9ec823\\harfbuzz-sys-0.1.15\\harfbuzz" "-G" "MSYS Makefiles" "-DCMAKE_INSTALL_PREFIX=D:\\programming\\git\\derin\\target\\debug\\build\\harfbuzz-sys-a5f7dd06f7f96198\\out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -m64" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -m64" "-DCMAKE_BUILD_TYPE=Debug"
-- Configuring incomplete, errors occurred!
 
--- stderr
error: linking with `gcc` failed: exit code: 1
  |
  = note: "gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\crt2.o" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.10ygozpgsu0ud2lu.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.15kq92zzbmxot4k9.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.16u6js6g0l3k1ic6.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.181cuta0v63atwcm.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.1im38lueib99jsk0.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.1kduva7sc7em934m.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.1mvmz58owquyropc.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.1y16o1qfye96o7m0.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.23dk5p5xq7wfpbjw.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.23tqyymcb18u96mb.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.2jqywn86b2gsqohu.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.2lyh15q6cjwzy18c.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.2qhkzqx5zqexj20y.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.3ayaeypdcro9d6yk.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.3rngp6bm2u2q5z0y.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.3wta9ctgdrpkmlpr.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.413jdks4e12oxna0.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.48721dc4k5qxei0u.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.49a7n47po4ttqjl7.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.49lx1q7cxvpykyv0.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.4ezmh1vbs95c5ack.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.4oc10dk278mpk1vy.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.4vp4wqj2v29i7mgy.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.4xq48u46a1pwiqn7.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.4yh8x2b62dcih00t.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.4ypvbwho0bu5tnww.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.56dly8q07ws8ucdq.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.57qy3vyd9bhiuaon.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.8xzrsc1ux72v29j.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.98g0d9x8aw3akpe.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.9elsx31vb4it187.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.b5mhfnazd92pmxo.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.spyjbt69vcsrx9q.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.y08g5q2x813c4wx.rcgu.o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.z9ox7biyn1otfln.rcgu.o" "-o" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.exe" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\harfbuzz_test-08995f4f600ae76a.crate.allocator.rcgu.o" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps" "-L" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\build\\servo-freetype-sys-0462d1271e152c62\\out/lib" "-L" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\build\\harfbuzz-sys-80e5021973efabc4\\out/lib" "-L" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\liblazy_static-6354ac3854bf518c.rlib" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\libfreetype-0ab2a748880f6e51.rlib" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\libfreetype_sys-7bdab5dc1192af88.rlib" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\libharfbuzz_sys-0f7380890dce59df.rlib" "D:\\programming\\git\\harfbuzz_test\\target\\debug\\deps\\liblibc-fefe43b2244c3859.rlib" "-Wl,--start-group" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libstd-49571416b3fd1c6f.rlib" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libpanic_unwind-8735c86531780c8c.rlib" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libunwind-959ba4391e10c630.rlib" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liblibc-52030e2cd68f7544.rlib" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc_system-4d28f73b9f75439b.rlib" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liballoc-2f5bf75e534c0dc8.rlib" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcore-0a758916232568ea.rlib" "-Wl,--end-group" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-65bef7a617529ebb.rlib" "-Wl,-Bdynamic" "-l" "advapi32" "-l" "ws2_32" "-l" "userenv" "-l" "shell32" "-Wl,-Bstatic" "-l" "gcc_eh" "-l" "pthread" "-Wl,-Bdynamic" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Users\\ossi\\.rustup\\toolchains\\stable-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: D:\programming\git\harfbuzz_test\target\debug\deps\libharfbuzz_sys-0f7380890dce59df.rlib(hb-ot-shape.cc.obj):hb-ot-shape.cc:(.xdata$_hb_ot_shaper_shape_plan_data_create+0x10): undefined reference to `__gxx_personality_seh0'
          D:\programming\git\harfbuzz_test\target\debug\deps\libharfbuzz_sys-0f7380890dce59df.rlib(hb-ot-layout.cc.obj):hb-ot-layout.cc:(.xdata$_ZNK2OT17CaretValueFormat315get_caret_valueEP9hb_font_t14hb_direction_tRKNS_14VariationStoreE+0xc): undefined reference to `__gxx_personality_seh0'
          D:\programming\git\harfbuzz_test\target\debug\deps\libharfbuzz_sys-0f7380890dce59df.rlib(hb-ot-layout.cc.obj):hb-ot-layout.cc:(.xdata$_ZNK2OT18SingleSubstFormat111would_applyEPNS_24hb_would_apply_context_tE+0xc): undefined reference to `__gxx_personality_seh0'
          D:\programming\git\harfbuzz_test\target\debug\deps\libharfbuzz_sys-0f7380890dce59df.rlib(hb-ot-layout.cc.obj):hb-ot-layout.cc:(.xdata$_ZNK2OT18SingleSubstFormat211would_applyEPNS_24hb_would_apply_context_tE+0xc): undefined reference to `__gxx_personality_seh0'
          D:\programming\git\harfbuzz_test\target\debug\deps\libharfbuzz_sys-0f7380890dce59df.rlib(hb-ot-layout.cc.obj):hb-ot-layout.cc:(.xdata$_ZNK2OT20MultipleSubstFormat111would_applyEPNS_24hb_would_apply_context_tE+0xc): undefined reference to `__gxx_personality_seh0'
          D:\programming\git\harfbuzz_test\target\debug\deps\libharfbuzz_sys-0f7380890dce59df.rlib(hb-ot-layout.cc.obj):hb-ot-layout.cc:(.xdata$_ZNK2OT21AlternateSubstFormat111would_applyEPNS_24hb_would_apply_context_tE+0xc): more undefined references to `__gxx_personality_seh0' follow
```

An example for this can be found in the `harfbuzz_test` repository below; if line 33 (with the offending `hb_face_create` call) is uncommented, the build proceeds through linking before `gcc` throws the given error. I discovered this on the latest nightly, but it also happens on the latest stable compiler.

The research I've done suggests that `cmake` isn't linking in `libstdc++-6.dll` properly, leading to it not being able to find the proper symbols. I don't know how you'd go about fixing this, though.

https://github.com/Osspial/harfbuzz_test/blob/master/src/main.rs#L32-L33

```
> rustc --version --verbose
rustc 1.27.1 (5f2b325f6 2018-07-07)
binary: rustc
commit-hash: 5f2b325f64ed6caa7179f3e04913db437656ec7e
commit-date: 2018-07-07
host: x86_64-pc-windows-gnu
release: 1.27.1
LLVM version: 6.0
```

--- a/harfbuzz-sys/build.rs
+++ b/harfbuzz-sys/build.rs
@@ -21,6 +21,9 @@ fn main() {
         let dst = cmake::Config::new("harfbuzz").build();
         println!("cargo:rustc-link-search=native={}/lib", dst.display());
         println!("cargo:rustc-link-lib=static=harfbuzz");
+        if target.contains("gnu") {
+            println!("cargo:rustc-link-lib=stdc++");
+        }
     } else {
         assert!(
             Command::new("make")

